test/utils/cartesian_hashtable_test.py

```diff
@@ -0,0 +1,296 @@
+        self.table.add((10, 10, 10))
+        pass
+
+
+@parameterized_class(("min_value", "max_value"), [
+    (0, 1000),
+    (-1000, 1000),
+    (-1000, 0)
+])
+class CartesianHashTableRandomTest(unittest.TestCase):
+
+    '''
+    Tests similar to above, but on three axes.
+    '''
+
+    def setUp(self):
+        '''
+        Set up simple table with `cartesian_tuple` as coordinates method.
+        '''
+        self.table = CartesianHashTable(cartesian_tuple)
+        # A hundred items, on three dimensions, with coordinates from 0 to 100.
+        self.dataset = [tuple(x) for x in numpy.random.randint(
+            self.min_value, self.max_value, (1000, 5))]
+
+    def test_size_zero_on_creation(self):
+        '''Size must be zero at creation.'''
+        self.assertEqual(self.table.size(), 0)
+
+    def test_add_some(self):
+        '''
+        Ensure that a bunch of items be added.
+        '''
+        for item in self.dataset:
+            self.table.add(item)
+        self.assertEqual(self.table.size(), len(self.dataset))
+
+    def test_remove_some(self):
+        '''
+        Ensure the item can be removed.
+        '''
+        for item in self.dataset:
+            self.table.add(item)
+        for item in self.dataset:
+            self.table.remove(item)
+        self.assertEqual(self.table.size(), 0)
+
+    def test_remove_value_error(self):
+        '''
+        Ensure ValueError is raised if trying to remove an item that isn't there.
+        '''
+        # Divide in two parts. From one third to two thirds go in the table, the rest stays out.
+        split = numpy.random.randint(len(self.dataset) // 3, (len(self.dataset) * 2) // 3)
+        inside = self.dataset[:split:]
+        outside = self.dataset[split::]
+
+        for i in inside:
+            self.table.add(i)
+
+        for o in outside:
+            self.assertRaises(ValueError, self.table.remove, o)
+
+    def test_contains(self):
+        '''
+        Ensure the contains method finds the inserted values.
+        '''
+        for item in self.dataset:
+            self.table.add(item)
+        for item in self.dataset:
+            self.assertIn(item, self.table)
+
+    def test_contains_if_not_contained(self):
+        '''
+        Ensure the contains method won't find a value that isn't there.
+        '''
+        for item in self.dataset:
+            self.table.add(item)
+        for item in self.dataset:
+            self.table.remove(item)
+
+        for item in self.dataset:
+            self.assertNotIn(item, self.table)
+
+    def test_to_scatter(self):
+        '''
+        Ensure the scatter lists contain only the inserted elements.
+        '''
+        for item in self.dataset:
+            self.table.add(item)
+        scatter = list(self.table.scatter())
+
+        # Convert dataset
+        expected = [list() for _ in self.dataset[0]]
+        for x in self.dataset:
+            for i in range(len(x)):
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=6277039)